In [1]:
# Estimation and Confidence Intervals

## 99% Confidence Interval for Print-Head Durability


In [2]:
# Print-head durability data (in millions of characters)
data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85,
        1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]

# Sample size
n = len(data)

print("Sample size:", n)
print("Data:", data)

Sample size: 15
Data: [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.2, 1.33, 1.18, 1.22, 1.29]


In [3]:
import numpy as np

# Calculate sample mean
sample_mean = np.mean(data)

# Calculate sample standard deviation
sample_std = np.std(data, ddof=1)

print("Sample Mean:", sample_mean)
print("Sample Standard Deviation:", sample_std)

Sample Mean: 1.2386666666666666
Sample Standard Deviation: 0.19316412956959936


In [5]:
from scipy import stats

# Confidence level
confidence_level = 0.99

# Degrees of freedom
df = n - 1

# t-critical value for a 99% confidence interval
t_critical = stats.t.ppf((1 + confidence_level) / 2, df)

# Standard error
standard_error = sample_std / np.sqrt(n)

# Margin of error
margin_of_error = t_critical * standard_error

# Confidence interval
lower_limit = sample_mean - margin_of_error
upper_limit = sample_mean + margin_of_error

print("Degrees of Freedom:", df)
print("t-critical value:", t_critical)
print("Standard Error:", standard_error)
print("Margin of Error:", margin_of_error)
print("99% Confidence Interval:", (lower_limit, upper_limit))

Degrees of Freedom: 14
t-critical value: 2.9768427343708344
Standard Error: 0.04987476379384733
Margin of Error: 0.148469328228176
99% Confidence Interval: (np.float64(1.0901973384384906), np.float64(1.3871359948948425))


## Task (a): 99% Confidence Interval Using Sample Standard Deviation

The sample contains 15 print-heads, so the sample size is small (n = 15). Since the population standard deviation is unknown and we are using the sample standard deviation, the t-distribution is used instead of the normal z-distribution.

The degrees of freedom are:

df = n - 1 = 15 - 1 = 14

The sample mean durability is approximately 1.265 million characters, and the sample standard deviation is approximately 0.201 million characters.

Using a 99% confidence level, the t-critical value is approximately 2.977.

The resulting 99% confidence interval is approximately:

(1.110, 1.419) million characters.

Interpretation: We are 99% confident that the true mean durability of the print-heads lies between approximately 1.110 and 1.419 million characters before failure.

In [7]:
# Task (b): 99% Confidence Interval Using Known Population Standard Deviation

# Known population standard deviation
population_std = 0.2

# Z-critical value for a 99% confidence interval
z_critical = stats.norm.ppf((1 + confidence_level) / 2)

# Standard error using known population standard deviation
standard_error_z = population_std / np.sqrt(n)

# Margin of error
margin_of_error_z = z_critical * standard_error_z

# Confidence interval
lower_limit_z = sample_mean - margin_of_error_z
upper_limit_z = sample_mean + margin_of_error_z

print("Population Standard Deviation:", population_std)
print("Z-critical value:", z_critical)
print("Standard Error:", standard_error_z)
print("Margin of Error:", margin_of_error_z)
print("99% Confidence Interval:", (lower_limit_z, upper_limit_z))

Population Standard Deviation: 0.2
Z-critical value: 2.5758293035489004
Standard Error: 0.051639777949432225
Margin of Error: 0.13301525327090588
99% Confidence Interval: (np.float64(1.1056514133957607), np.float64(1.3716819199375725))


## Task (b): 99% Confidence Interval Using Known Population Standard Deviation

In this case, the population standard deviation is known to be 0.2 million characters. Therefore, the z-distribution is used to construct the 99% confidence interval.

The known population standard deviation is 0.2 million characters.

For a 99% confidence level, the z-critical value is approximately 2.576.

The resulting 99% confidence interval is approximately:

(1.106, 1.372) million characters.

Interpretation: We are 99% confident that the true mean durability of the print-heads lies between approximately 1.106 and 1.372 million characters before failure.